### DISCUSSION TITLE
File Downlink cycleTime and Run Synchronization

### DISCUSSION THREAD

**Discussion by teddylambert** [nasa/fprime#1957] [Q&A] [answered]
I am using the fileDownlink component in a deployment, and am curious about how the `cycleTime` argument setting vs run port rate group synchronization works. From the code, it looks like cycleTime simply gets added to a counter when the `run` port gets called. Does this mean that synchronization is up to the developer? For example, if I have the run port on a 10Hz rate group, I need to manually set CYCLE_TIME to 100?

**Comment by timcanham** [comment, marked as answer]
@teddylambert That is correct. There are timeouts/cooldowns set via the `configure()` function, and the `Run` handler is used to track expiration, so the `cycleTime` argument is used to tell the component how much time passes between each invocation of the `Run` handler. Hypothetically we could have used system time calls, but we wanted it to be portable to embedded systems that might not support those calls.
